share/spack/templates/container/amazonlinux_2.dockerfile

```diff
@@ -0,0 +1,27 @@
+{% extends "container/bootstrap-base.dockerfile" %}
+{% block install_os_packages %}
+RUN yum update -y \
+ && yum groupinstall -y "Development Tools" \
+ && yum install -y \
+        curl \
+        findutils \
+        gcc-c++ \
+        gcc \
+        gcc-gfortran \
+        git \
+        gnupg2 \
+        hostname \
+        iproute \
+        Lmod \
+        make \
+        patch \
+        python \
+        python-pip \
+        python-setuptools \
+        tcl \
+        unzip \
+        which \
```

Comment:
where do we actually call `which` in spack scripts?  I think we should start preferring `command -v`, as that's actually a builtin.  Can we change that instead of including this in images?

Reply:
I'll check what can be trimmed, but for reference I started from:

https://github.com/spack/spack/blob/6b12000a7c81ca26b874b087a1a3c88034cfdd34/share/spack/docker/centos-7.dockerfile#L16-L34

which is what we currently build on Dockerhub

Reply:
I'll check what can be trimmed, but for reference I started from:

https://github.com/spack/spack/blob/6b12000a7c81ca26b874b087a1a3c88034cfdd34/share/spack/docker/centos-7.dockerfile#L16-L34

which is what we currently build on Dockerhub

Reply:
See ab4533c